### ISSUE TITLE
Bug: default route for WebApi is wrong

### ISSUE THREAD

**Issue by cbordeman** [NSwag/NSwag#287]
This is a bug I think.  In WebApiToSwaggerGeneratorSettings.cs:

`public string DefaultUrlTemplate { get; set; } =  "api/{controller}/{action}/{id}";`

This is the default route for **MVC**, not **WEBAPI**.  The result is you must add the following attribute to all your api controller **classes** (not methods):

`[Route("api/{controller}/{id}")]`
(the correct default for WebAPI)

I had to spend a few hours getting stepping through your code to figure this out.  I know the Readme.md briefly mentions that the Route attribute is supported, but I doubt most users got that they needed to do this.  I personally used the NSwag Studio tool and was confused by the result.

Also, can we get some documentation on how to have our models, client interfaces, and concrete client classes generated separately, or at least how to provide our own models and have those not generated?    I could separate them manually but that's a ton of work during initial development to do every time I make a small change to a model.  Thanks!

P.S. Awesome tool!!

**Comment by rsuter** [comment]
Cant you just change the default template in the generator settings? And yes, if the template is without {action} we should fix that (even if it may be a breaking change)...

**Comment by cbordeman** [comment]
I totally didn't see that option in the UI, thanks for pointing it out.  As for a breaking change, if you're saving the value in the .nswag file, it wouldn't be a breaking change except for people who reenter all their parameters every time they generate.
Again, this tool is simply amazing, thanks for the mountains of work you did!  One small complaint, some documentation would be quite helpful, especially around the operation generation modes.  This is obvious to me now, but only after seeing the code.

Date: Sun, 11 Sep 2016 10:27:06 -0700

**Comment by rsuter** [comment]
NSwag can also be used in cmd line mode and this will use the default route template and changing it is a breaking change in this scenario...

**Comment by rsuter** [comment]
Documentation on operation generation modes: https://github.com/NSwag/NSwag/wiki/ClientGeneratorBaseSettings

**Comment by rsuter** [comment]
Ok, just checked this. Created a new Web project and the default Web API route is indeed wrong:

**Comment by rsuter** [comment]
Please add a new issue for the separate model/interface/client generation... (currently you can only separate interface+model and client classes

**Comment by rsuter** [comment]
Is this enough?

**Comment by cbordeman** [comment]
Yes, thanks.

**Comment by cbordeman** [comment]
Should note that path segment based client generation doesn't work with this default path because the action isn't present, and the path based extraction code expects it to be.  I'm not sure what the default mode is.

**Comment by rsuter** [comment]
The default is from operation id... So its fine

**Comment by rsuter** [comment]
Btw: By path also uses the method name as fallback...

Can this be closed?
